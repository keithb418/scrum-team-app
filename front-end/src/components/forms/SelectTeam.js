import React from "react";
import { FormGroup, ControlLabel, FormControl } from "react-bootstrap";
import PropTypes from "prop-types";

const SelectTeam = ({ teams = [], selected = "", onSelect }) => {
  return (
    <FormGroup controlId="teamSelect">
      <ControlLabel>Team<sup><i className="fa fa-asterisk required"></i></sup></ControlLabel>
      <FormControl
        name="team"
        defaultValue={selected}
      	componentClass="select"
      	placeholder="Select Team"
      	onChange={onSelect} >

        <option>Select Team</option>

        {teams.map(team =>
          <option
            value={team._id}
            key={team._id}>{team.name}</option>
        )}

      </FormControl>
    </FormGroup>
  );
};

SelectTeam.propTypes = {
  onSelect: PropTypes.func,
  teams: PropTypes.array,
};

export default SelectTeam;