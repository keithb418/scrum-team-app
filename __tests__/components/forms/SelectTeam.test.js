import SelectTeam from "../../../front-end/src/components/forms/SelectTeam";
import React from "react";
import { shallow } from "enzyme";
import teamsData from '../../fixtures/teams';

let onSelect, wrapper
test('should render the SelectTeam component correctly', () => {
    onSelect = jest.fn();
    wrapper = shallow(
      <SelectTeam 
          teams={teamsData} 
          onSelect={onSelect}/>
    );
  expect(wrapper).toMatchSnapshot();
});