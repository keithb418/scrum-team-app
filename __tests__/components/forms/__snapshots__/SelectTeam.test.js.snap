// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the SelectTeam component correctly 1`] = `
<FormGroup
  bsClass="form-group"
  controlId="teamSelect"
>
  <ControlLabel
    bsClass="control-label"
    srOnly={false}
  >
    Team
    <sup>
      <i
        className="fa fa-asterisk required"
      />
    </sup>
  </ControlLabel>
  <FormControl
    bsClass="form-control"
    componentClass="select"
    defaultValue=""
    name="team"
    onChange={[MockFunction]}
    placeholder="Select Team"
  >
    <option>
      Select Team
    </option>
    <option
      key="1"
      value="1"
    >
      Team 1
    </option>
    <option
      key="2"
      value="2"
    >
      Team 2
    </option>
  </FormControl>
</FormGroup>
`;
